workflows/eager.nf

```diff
@@ -296,13 +268,42 @@ workflow EAGER {
     //
 
     if ( !params.skip_qualimap ) {
-       ch_qualimap_input = ch_dedupped_bams
+        ch_snp_capture_bed = REFERENCE_INDEXING.out.snp_capture_bed
+            .map{
+                WorkflowEager.addNewMetaFromAttributes( it, "id" , "reference" , false )
+            }
+        ch_qualimap_input = ch_dedupped_bams
             .map {
             meta, bam, bai ->
                 [ meta, bam ]
             }
-        QUALIMAP_BAMQC(ch_qualimap_input, ch_snpcapture_bed)
-        ch_versions      = ch_versions.mix( QUALIMAP_BAMQC.out.versions )
+            .map {
+                WorkflowEager.addNewMetaFromAttributes( it, "reference" , "reference" , false )
+            }
+            .combine(
+                by: 0,
+                ch_snp_capture_bed
+            )
+            .branch {
+                ignore_meta, meta, bam, meta2, snp_capture_bed ->
+                withbed: snp_capture_bed != ""
+                nobed: true
+            }
+        ch_qualimap_input_with = ch_qualimap_input.withbed
+            .multiMap{
+                ignore_meta, meta, bam, meta2, snp_capture_bed ->
+                bam:             [ meta, bam ]
+                snp_capture_bed: [ snp_capture_bed ]
+            }
+        QUALIMAP_BAMQC_WITHBED( ch_qualimap_input_with.bam, ch_qualimap_input_with.snp_capture_bed )
+        ch_qualimap_input_without = ch_qualimap_input.nobed
+            .map{
+                ignore_meta, meta, bam, meta2, snp_capture_bed ->
+                [ meta, bam ]
+            }
+        QUALIMAP_BAMQC_NOBED( ch_qualimap_input_without, [] )
+        ch_qualimap_output = QUALIMAP_BAMQC_WITHBED.out.results.mix( QUALIMAP_BAMQC_NOBED.out.results )
```

Comment:
Do we have a MultiQC channel yet?

Reply:
for qualimap? Yes it's mixed in, if skip_qualimap is false (see line 559).